Dependencies/Authentication/Sources/Authentication/Helpers/SignInFacebookHelper.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  SignInFacebookHelper.swift
+//  
+//
+//  Created by Kamil Wójcicki on 07/10/2023.
+//
+
+import Foundation
+import FBSDKLoginKit
+
+public struct FacebookSignInResultModel {
+    let accessToken: String
+    
+    public init(accessToken: String) {
+        self.accessToken = accessToken
+    }
+}
+
+public final class SignInFacebookHelper {
+    
+    @MainActor
+    func signIn() async throws -> FacebookSignInResultModel {
+
+        let manager = LoginManager()
+        let loginResult = try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<FacebookSignInResultModel, Error>) in
+            manager.logIn(permissions: ["public_profile", "email"], from: nil) { result, error in
+                if let error = error {
+                    continuation.resume(throwing: error)
+                } else if let result = result, !result.isCancelled {
+                    guard let accessToken = AccessToken.current?.tokenString else { return }
+
+                    let signInResult = FacebookSignInResultModel(accessToken: accessToken)
+                    continuation.resume(returning: signInResult)
+                } else {
+                    continuation.resume(throwing: NSError(domain: "Facebook", code: 0, userInfo: [NSLocalizedDescriptionKey: "Facebook login was cancelled."]))
+                }
+            }
+        }
+        return loginResult
+    }
+}
```

Comment:
Internal class - bo nie używasz jej poza paczką

`@MainActor` - jest używany gdy chcesz zrobić updateUI ta funkcja nie wpływa bezpośrednio na UI wiec nie trzeba ją oznaczać jako `@MainActor`. 


*Dla lepszego zrozumienia.
W public final class SocialMediaButtonViewModel: ObservableObject { ... } masz funkcje 
`private func signInWithFacebook() async throws { ... }` I ona I to jej reultat wpływa bezpośrednio na UI twojego widoku wiec to ona powinna być oznaczona jako `@MainActor`. 


`continuation.resume(throwing: NSError(domain: "Facebook", code: 0, userInfo: [NSLocalizedDescriptionKey: "Facebook login was cancelled."]))`

Nie jestem pewien czy to jest konieczne a jeśli jest - to czy jest poprawnie to zrobione